File: tools/bench/answers/go_haiku/s1/06_stats.go
package main

import "fmt"

func main() {
	nums := []float64{4, 8, 15, 16, 23}
	min := nums[0]
	max := nums[0]
	sum := 0.0
	for _, num := range nums {
		if num < min {
			min = num
		}
		if num > max {
			max = num
		}
		sum += num
	}
	mean := sum / float64(len(nums))
	fmt.Println(int(min))
	fmt.Println(int(max))
	fmt.Println(mean)
}

File: tools/bench/answers/rust_haiku/s3/06_stats.rs
fn main() {
    let list = [4, 8, 15, 16, 23];
    let mut min = list[0];
    let mut max = list[0];
    let mut sum = 0;

    for &num in &list {
        if num < min {
            min = num;
        }
        if num > max {
            max = num;
        }
        sum += num;
    }

    let mean = sum as f64 / list.len() as f64;

    println!("{}", min);
    println!("{}", max);
    println!("{}", mean);
}
